### DISCUSSION TITLE
How can I use GroupKFold cross-validation with scikit-survival?

### DISCUSSION THREAD

**Discussion by DanielWTang** [sebp/scikit-survival#251] [Q&A] [answered]
Hello All, 

I have recently found about the scikit-survival package and I am a relatively new python user (mostly working on R). 

I am trying to implement the CoxnetSurvivalAnalysis on my dataset using cross-validation (keeping prediction from each fold). And since my dataset contains various time points from the same individuals, I'd like to do a GroupKFold to make sure no data from the same individual are used in both the training and testing datasets. However, I wasn't able to find anything in the User's guide specifying the use of group cross-validation.

I suppose I can write an external loop to split my data and implement the group CV that way. However, since I am interested in the sksurv.meta.Stacking function later on, I am not sure a work around is the best option here. 

Has anyone worked with similar projects before? Any shared experience is highly appreciated! 

Thank you

**Comment by sebp** [comment, marked as answer]
There's nothing special about it with scikit-survival, just follow the 
scikit-learn guide and train/test CoxnetSurvivalAnalysis instead of one 
of scikit-learn's models.

See 
https://scikit-learn.org/stable/modules/cross_validation.html#group-k-fold

**Reply by DanielWTang** [reply]
Thanks a lot for your anwser!

I am interested in the sksurv.meta.Stacking function but I can't seem to find an example usage anywhere in the user's guide. I am wondering if you can please share some scripts/code as an example for that?

Thanks again for your time and help!

**Comment by sebp** [comment]
Have a look at 
https://github.com/sebp/scikit-survival/blob/9ed69b422e7b4919383a8ace22291cfc47ea3ac0/tests/test_stacking.py#L64
